db/mongo/client.go

```diff
@@ -0,0 +1,136 @@
+	log().Infof("initializing mongodb: %s", uri)
+
+	cli, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
+	if err != nil {
+		return nil, wraperrors.Wrap(err, "failed to connect")
+	}
+
+	return &client{conn: cli}, nil
+}
+
+// Registration stores new user details in 'user' collection
+func (m client) Registration(userInfo *models.User) error {
+	_, err := m.ViewProfile(userInfo.Email)
+	// add unregistered users only
+	if err != nil {
+		if userInfo.ID != "" {
+			return nil
+		}
+		userInfo.ID = guuid.NewV4().String()
+		collection := m.conn.Database(viper.GetString(config.DbName)).Collection(userCollection)
+		if _, err := collection.InsertOne(context.TODO(), userInfo); err != nil {
+			return wraperrors.Wrap(err, "failed to add user")
+		}
+
+		return nil
+	}
+
+	return err
+}
+
+// UserLogin return password from db on successful call else error
+func (m client) UserLogin(email string) (string, error) {
+	var user *models.User
+	collection := m.conn.Database(viper.GetString(config.DbName)).Collection(userCollection)
+	if err := collection.FindOne(context.TODO(), bson.M{"email": email}).Decode(&user); err != nil {
+		if mongo.ErrNoDocuments != nil {
+			return "", wraperrors.Wrap(err, "failed to fetch user....not found")
+		}
+
+		return "", err
+	}
+
+	return user.Password, nil
+}
+
+// ViewProfile return user details from db on successful call else error
+func (m client) ViewProfile(email string) (*models.User, error) {
+	var user *models.User
+	collection := m.conn.Database(viper.GetString(config.DbName)).Collection(userCollection)
+	if err := collection.FindOne(context.TODO(), bson.M{"email": email}).Decode(&user); err != nil {
+		if mongo.ErrNoDocuments != nil {
+			return nil, wraperrors.Wrap(err, "failed to fetch user....not found")
+		}
+
+		return nil, err
+	}
+
+	return user, nil
+}
+
+// SubscribeItem add user subscribed item details in the 'subscription' collection
+func (m client) SubscribeItem(email string, subscriptionInfo *models.Subscription) error {
+	if subscriptionInfo.ID != "" {
+		return nil
+	}
+
+	subscriptionInfo.ID = guuid.NewV4().String()
+	user, err:= m.ViewProfile(email)
+	if err != nil {
+		return wraperrors.Wrap(err, "failed to retrieve user details")
+	}
+	subscriptionInfo.UserId = user.ID
+
+	collection := m.conn.Database(viper.GetString(config.DbName)).Collection(subscriptionCollection)
+	if _, err := collection.InsertOne(context.TODO(), subscriptionInfo); err != nil {
+		return wraperrors.Wrap(err, "failed to add subscription details")
+	}
+
+	return nil
+}
+
+// ViewSubscription return slice of given user subscriptions on successful call
+func (m client) ViewSubscription(email string) ([]*models.Subscription, error) {
+	user, err := m.ViewProfile(email)
+	if err != nil {
+		return nil, wraperrors.Wrap(err, "failed to add subscription details")
+	}
+	collection := m.conn.Database(viper.GetString(config.DbName)).Collection(subscriptionCollection)
```

Comment:
add spaces to separate code logic for better readability.

Reply:
Didn't get your point.